fix: Scroll lock when transitioning on mobile via slide-over nav

filipewl opened the pull request:
## What's the purpose of this pull request?

After the changes from https://github.com/vtex-sites/base.store/pull/329, a bug was accidentally introduced when navigating using one of the links inside the mobile slide-over navigation: scrolling was locked because the `.no-scroll` CSS class was kept in the `#layout` element. The `.no-scroll` is added when the slide-over is opened to prevent scrolling the content on the background. This happened because the links have been replaced by Gatsby's so there isn't a full page refresh anymore.

## How does it work?

The fix works by removing the `.no-scroll` CSS class when needed.

[Before](https://sfj-2ce7b26--base.preview.vtex.app/)|[After](https://preview-344--base.preview.vtex.app/)
-|-
https://user-images.githubusercontent.com/381395/154371235-2d34df1c-4c51-47b1-9fda-5780b74ce4af.mp4|https://user-images.githubusercontent.com/381395/154371303-e37a4f7d-8689-46c4-a508-7452c895f791.mp4

## How to test it?

https://preview-344--base.preview.vtex.app/

1. On mobile viewport, visit a page like the [Home](https://preview-344--base.preview.vtex.app/).
2. Open the slide-over navigation and click on any of the collection/PLP/categories links.
3. Upon arriving on the new page, you should be able to scroll freely.

## References

- https://vtex-dev.atlassian.net/browse/FSSS-201

Commit:
Remove the `.no-scroll` class when unmounting

Commit:
Add entry to the changelog

Commit:
Merge branch 'master' into fix/FSSS-201-mobile-no-scroll

eduardoformiga reviewed (approved):
LGTM! 👏 🎉

filipewl commented on src/components/ui/SlideOver/SlideOver.tsx:
```diff
@@ -56,6 +56,10 @@ const SlideOver = ({
     }
+
+    return () => {
+      layout.current?.classList?.remove('no-scroll')
```
Now I'm not happy about where I placed this... This component has 3 `useEffect`s, but adding this to the 1st one doesn't fix the issue, while it works for the 2nd and 3rd ones. Thinking where the is best logical place to place this.
🤔 

<img width="559" alt="CleanShot 2022-02-16 at 20 30 46@2x" src="https://user-images.githubusercontent.com/381395/154375405-cfebd020-d325-4639-afc8-b4cbb3fbc8a9.png">

hellofanny replied:
Hmm. Actually, I think it's okay to use multiple effects here. 🤔 

As suggested by the documentation: [Use Multiple Effects to Separate Concerns](https://reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns)

hellofanny replied:
Btw, I don't think to verify `handleClose` (line 66). Can you check it again and then remove it pls ✨

lucasfp13 replied:
I also think it's okay to leave on that `useEffect` as it's related to the `layout`.

I believe it didn't work for the first effect because when the cleanup function is called the component is already on unmount state, possibly without the current `layout` value, or possibly it has changed. You can try to store the `layout.current` value inside the `useEffect` scope to keep the value (see this discussion [here](https://stackoverflow.com/questions/67069827/cleanup-ref-issues-in-react))

hellofanny commented on CHANGELOG.md:
```diff
@@ -27,6 +27,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - React tree re-rendering
 - Footer rendering pipeline
+- Scroll lock when transitioning pages on mobile via slide-over navigation
```
```suggestion
- Scroll lock when transitioning pages on mobile via `SlideOver` component navigation
```

hellofanny reviewed (approved):
Nice fix! ✨

filipewl committed:
Improve wording for in the changelog

Co-authored-by: Fanny Chien <fannychienn@gmail.com>

israelswf reviewed (approved):
✨

Commit:
Remove conditional for checking `handleClose` since it's truthy

> I'm not sure if we need this if here since handleClose is a function and is always true. 👀
https://github.com/vtex-sites/base.store/pull/262#discussion_r794942711

> Btw, I don't think to verify handleClose (line 66). Can you check it again and then remove it pls ✨
https://github.com/vtex-sites/base.store/pull/344/files#r808602642

Co-authored-by: Fanny Chien <fannychienn@gmail.com>

Commit:
Keep the current pattern for checking if `layout.current` is truthy

tlgimenes reviewed (approved):
nice!

filipewl committed:
Merge branch 'master' into fix/FSSS-201-mobile-no-scroll
